zrml/prediction-markets/src/migrations.rs

```diff
@@ -0,0 +1,336 @@
+use crate::{Config, MarketIdsPerOpenBlock, MarketIdsPerOpenTimeFrame, Pallet};
+use frame_support::{
+    dispatch::Weight,
+    log,
+    pallet_prelude::PhantomData,
+    traits::{Get, OnRuntimeUpgrade, StorageVersion},
+};
+use zeitgeist_primitives::types::{MarketPeriod, MarketStatus, PoolStatus};
+use zrml_market_commons::MarketCommonsPalletApi;
+
+const SWAPS_REQUIRED_STORAGE_VERSION: u16 = 2;
+const SWAPS_NEXT_STORAGE_VERSION: u16 = 3;
+const PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION: u16 = 2;
+const PREDICTION_MARKETS_NEXT_STORAGE_VERSION: u16 = 3;
+
+pub struct MigrateMarketPoolsBeforeOpen<T>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for MigrateMarketPoolsBeforeOpen<T> {
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        let mut total_weight = T::DbWeight::get().reads(1);
+        if utility::get_on_chain_storage_version_of_swaps_pallet() != SWAPS_REQUIRED_STORAGE_VERSION
+        {
+            log::info!("Skipping storage migration of market pools; swaps already up to date");
+            return total_weight;
+        }
+        if StorageVersion::get::<Pallet<T>>() != PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION {
+            log::info!(
+                "Skipping storage migration of market pools; prediction-markets already up to date"
+            );
+            return total_weight;
+        }
+        log::info!("Starting storage migration of market pools");
+
+        let current_block = <frame_system::Pallet<T>>::block_number();
+        let current_time_frame =
+            Pallet::<T>::calculate_time_frame_of_moment(T::MarketCommons::now());
+        total_weight = total_weight.saturating_add(T::DbWeight::get().reads(2));
+
+        for (market_id, market) in T::MarketCommons::market_iter() {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+            if market.status != MarketStatus::Active {
+                continue;
+            }
```

Comment:
You could inject this constraint into the iteration using the functional programming style function [`filter`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter).

Reply:
Mitigated in ff241d88d9e2f51d76b2a9aa8cd8368671245e23